### PR TITLE
[DPE-4173] Miscellaneous improvements to continue stabilizing exporter integration tests

### PR DESCRIPTION
**Pull request by shayancanonical**
## Issue
1. We are using connection pools in our exporter integration tests. As these connection pools use connections with keep-alive, the guess is that the server has connections in `TIME_WAIT` for a period of time when it shuts down. Immediately starting the server (in consecutive tests) results in `bind-address in use` error 
2. We are running to 2 possibly flakey tests back to back (the first one unrelating and the second one shortly relating with COS right after)

## Solution
1. Use `requests.get(stream=False)` instead of using connection pools
2. Split the 2 tests into separate files so that they are part of separate groups and run separately

## Notes
Tests passed in 1 retry compared to usually higher retries

### CODE REVIEW

**Review by taurus-forever — APPROVED** [review]
Interesting approach!

**Comment by shayancanonical** [comment]
It is by no means a call to surrender! Plan on continuing investigating the connection in TIME_WAIT with upstream: https://github.com/rluisr/mysqlrouter_exporter/issues/67

File: tests/integration/test_exporter.py [hunk 1]
```diff
@@ -120,17 +109,13 @@ async def test_exporter_endpoint(ops_test: OpsTest) -> None:
     unit_address = await get_unit_address(ops_test, unit.name)
 
     try:
-        http.request("GET", f"http://{unit_address}:49152/metrics")
-    except urllib3.exceptions.MaxRetryError as e:
-        assert (
-            "[Errno 111] Connection refused" in e.reason.args[0]
-        ), "❌ expected connection refused error"
+        with requests.Session() as session:
+            session.get(f"http://{unit_address}:49152/metrics", stream=False)
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/test_exporter.py:113]: the usage of `requests.Session()` was an attempt to explicitly close the session after every request. will use `requests.get()` as it is easier to understand

File: tests/integration/test_exporter_with_tls.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+MYSQL_APP_NAME = MYSQL_DEFAULT_APP_NAME
+MYSQL_ROUTER_APP_NAME = MYSQL_ROUTER_DEFAULT_APP_NAME
+APPLICATION_APP_NAME = APPLICATION_DEFAULT_APP_NAME
+GRAFANA_AGENT_APP_NAME = "grafana-agent-k8s"
+SLOW_TIMEOUT = 25 * 60
+RETRY_TIMEOUT = 3 * 60
+
+if juju_.is_3_or_higher:
+    TLS_APP_NAME = "self-signed-certificates"
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/test_exporter_with_tls.py:36]: why are they not constants? i believe they are as their values are not changing

also, all such variables are constants in tests, so for now, for consistency, i would prefer to leave as is

File: src/rock.py [hunk 3]
```diff
@@ -195,7 +195,7 @@ def update_mysql_router_exporter_service(
         )
         self._container.add_layer(self._EXPORTER_SERVICE_NAME, layer, combine=True)
         # `self._container.replan()` does not stop services that have been disabled
-        # Use `restart()` and `stop()` instead
+        # Explicitly use `stop()` instead
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply src/rock.py]: reverted in [da0de81](https://github.com/canonical/mysql-router-k8s-operator/pull/254/commits/da0de8120776b2e369f6bdead54ac0d616d21756)

File: tests/integration/test_exporter.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/test_exporter.py]: use `requests.get()` in [da0de81](https://github.com/canonical/mysql-router-k8s-operator/pull/254/commits/da0de8120776b2e369f6bdead54ac0d616d21756)

File: tests/integration/test_exporter.py [hunk 4]
```diff
@@ -247,24 +135,18 @@ async def test_exporter_endpoint_with_tls(ops_test: OpsTest) -> None:
...
-        mysql_router_unit.name,
-        host="127.0.0.1",
-        port=6446,
-    )
-    assert "CN = Test CA" in issuer, f"Expected mysqlrouter certificate from {TLS_APP_NAME}"
+            with requests.Session() as session:
+                response = session.get(f"http://{unit_address}:49152/metrics", stream=False)
+                assert (
+                    response.status_code == 200
+                ), "❌ cannot connect to metrics endpoint with relation with cos"
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/test_exporter.py]: replaced in [da0de81](https://github.com/canonical/mysql-router-k8s-operator/pull/254/commits/da0de8120776b2e369f6bdead54ac0d616d21756)

File: tests/integration/test_exporter.py [hunk 5]
```diff
@@ -247,24 +135,18 @@ async def test_exporter_endpoint_with_tls(ops_test: OpsTest) -> None:
...
+                assert "mysqlrouter_route_health" in str(
+                    response.text
+                ), "❌ did not find expected metric in response"
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/test_exporter.py]: addressed in [da0de81](https://github.com/canonical/mysql-router-k8s-operator/pull/254/commits/da0de8120776b2e369f6bdead54ac0d616d21756)

File: tests/integration/test_exporter_with_tls.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulomach) [reply tests/integration/test_exporter_with_tls.py:36]: color of the buttons, as someone like to says. let's move forward

**Review by paulomach — APPROVED** [review]

**Pull request merged**